expr_checker.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import ast
+from copy import copy
+
+
+_safe_type = None
+_readonly = None
+
+
+def check_type(obj):
+    if type(obj) in {list, tuple, set}:
+        list(map(check_type, obj))
+        return obj
+    elif type(obj) not in _safe_type:
+        raise ValueError(f"safe_eval didn't like {obj}")
+    else:
```

Comment:
It's needed when I call it in the AST like that:

```python
        if isinstance(node.ctx, ast.Load):
            return ast.Call(
                func=ast.Name("check_type", ctx=ast.Load()),
                args=[node],
                keywords=[]
            )
```

It will returns the given argument and pass it to the called function. If you want I can remove it and put the `return` outside of the conditional statement, but I think it's make it more readable.

Reply:
I means you can just do
```
return value
```

instead of
```
else:
    return value
```